spec/models/mail_server_log_spec.rb

```diff
@@ -331,21 +331,56 @@
 
   describe '#delivery_status' do
 
-    context 'using the :exim MTA' do
+    context 'if there is a stored value' do
+      let(:log) do
+        MailServerLog.new(:line => "log text",
+                          :delivery_status => 'bounce_arrival')
+      end
+
+      it 'returns the stored value' do
+        status = MailServerLog::EximDeliveryStatus.new(:bounce_arrival)
+        expect(log.delivery_status).to eq(status)
+      end
+
+      it 'does not look at the line text' do
+        expect(log).to_not receive(:line)
+        log.delivery_status
+      end
+
+    end
+
+    context 'there is not a stored value' do
+
+      it 'attempt to parse the line text' do
```

Comment:
this may need more detail - I was thinking 'attempts' as it might not find a status value, although it's still parsing it either way